packages/react/src/analytics/integrations/Forter/__tests__/Forter.test.ts

```diff
@@ -0,0 +1,209 @@
+  resolve => (mockPostTrackingsPromiseResolve = resolve),
+);
+const mockSiteId = 'c3f674e6511e';
+const mockOrigin = 'Whitelabel Portal';
+const mockForterToken =
+  '44807307964b4931a89bf364e580bdc5_1594724271711__UDF43_9ck';
+const mockForterTokenLoadedPromise = Promise.resolve(mockForterToken);
+
+// @ts-expect-error
+const strippedDownAnalytics: StrippedDownAnalytics = {
+  createEvent: (type: TrackTypes) =>
+    Promise.resolve({ ...loadIntegrationData, type }),
+};
+
+jest.mock('@farfetch/blackout-client', () => ({
+  ...jest.requireActual('@farfetch/blackout-client'),
+  postTracking: jest.fn(async () => {
+    mockPostTrackingsPromiseResolve();
+    return await mockPostTrackingsPromise;
+  }),
+}));
+
+jest.mock(
+  '../forterTokenLoadedEventDetector',
+  () =>
+    class {
+      getForterToken = jest.fn(async () => await mockForterTokenLoadedPromise);
+    },
+);
+
+utils.logger.warn = jest.fn();
+
+describe('Forter', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should return true on its shouldLoad method', () => {
+    expect(Forter.shouldLoad()).toBe(true);
+  });
+
+  it('should return an instance of Forter class in its createInstance method', () => {
+    expect(
+      Forter.createInstance(
+        { siteId: mockSiteId },
+        loadIntegrationData,
+        strippedDownAnalytics,
+      ),
+    ).toBeInstanceOf(Forter);
+  });
+
+  it('should return an error when siteId is not set on Forter Options.', () => {
+    // @ts-expect-error try loading forter without passing siteId
+    Forter.createInstance({}, loadIntegrationData, strippedDownAnalytics);
+
+    expect(utils.logger.warn).toHaveBeenCalledWith(
+      "[Forter] - `siteId` parameter was not provided in options. It's mandatory to load Forter Integration.",
+    );
+  });
+
+  it('should log a warn message if `origin` is not specified in options', () => {
+    Forter.createInstance(
+      { siteId: mockSiteId },
+      loadIntegrationData,
+      strippedDownAnalytics,
+    ) as Forter;
+
+    expect(utils.logger.warn).toHaveBeenCalledWith(
+      "[Forter] - `origin` parameter was not provided in options. It's advisable to provide an origin option to aid in debugging",
+    );
+  });
+
+  it('should post a message to Omnitracking with forter token when it is loaded', async () => {
+    const createEventMock = async function (
+      type: TrackTypesValues,
+      event: string,
+    ) {
+      return Promise.resolve({
+        ...eventData,
+        event,
+        type,
+      });
+    };
+
+    const forterInstance = Forter.createInstance(
+      {
+        siteId: mockSiteId,
+        origin: mockOrigin,
+      },
+      loadIntegrationData,
+      { ...strippedDownAnalytics, createEvent: createEventMock },
+    );
+    // @ts-expect-error accessing private property just for testing
+    const spy = jest.spyOn(forterInstance, 'resolveOnSetUserPromise');
+    // @ts-expect-error test data
+    const data: SetUserEventData = {};
```

Comment:
There is an `onSetUserEventData` mock you can import and use here. Using an event that is not correctly mocked can skew the tests. Change it to be like this, please and you can then remove the `@ts-expect-error` from the line before:

```suggestion
    const data: SetUserEventData = onSetUserEventData;
```